src/main/java/ch/hevs/cloudio/endpoint/JobsManager.java

```diff
@@ -0,0 +1,141 @@
+package ch.hevs.cloudio.endpoint;
+
+import org.eclipse.paho.client.mqttv3.MqttAsyncClient;
+import org.eclipse.paho.client.mqttv3.MqttException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.List;
+
+public class JobsManager {
+
+    private static final Logger log = LoggerFactory.getLogger(JobsManager.class);
+
+    private enum CmdJobs{
+        listJobs;
+    }
+
+    private JobsManager(){
+    }
+
+    private static JobsManager instance = new JobsManager();
+
+    public static JobsManager getInstance(){
+        return instance;
+    }
+
+    public List<String> listScripts(String filePath){
+        File folder = new File(filePath);
+        File[] listOfFiles = folder.listFiles();
+        ArrayList result = new ArrayList();
+
+        for (File listOfFile : listOfFiles) {
+            if (listOfFile.isFile()) {
+                result.add(listOfFile.getName());
+            }
+        }
+        return result;
+    }
+
+    public void executeJob(String jobURI, String filePath, Boolean output, MqttAsyncClient mqtt, String uuid){
+        ProcessBuilder processBuilder = new ProcessBuilder();
+
+        String scheme = jobURI.split("://")[0];
```

Comment:
If you save the split operations result, you do not need to split the string twice and you can check the actual number of components.

Reply:
Yes, will change that